gene2phenotype_project/gene2phenotype_app/views/locus_genotype_disease.py

```diff
@@ -218,6 +219,150 @@ def update(self, request, stable_id):
         else:
             return Response({"errors": serializer.errors}, status=status.HTTP_400_BAD_REQUEST)
 
+class LGDUpdateMechanism(BaseUpdate):
+    http_method_names = ['patch', 'options']
+    serializer_class = MolecularMechanismSerializer # TO CHECK
+    permission_classes = [permissions.IsAuthenticated]
+
+    def get_queryset(self):
+        """
+            Retrieves a queryset of LocusGenotypeDisease objects associated with a stable ID
+            for the authenticated user.
+
+            Args:
+                stable_id (str): The stable ID from the URL kwargs.
+
+            Returns:
+                QuerySet: A queryset of LocusGenotypeDisease objects.
+
+            Raises:
+                Http404: If the stable ID does not exist.
+                PermissionDenied: If update is not allowed.
+        """
+        stable_id = self.kwargs['stable_id']
+
+        g2p_stable_id = get_object_or_404(G2PStableID, stable_id=stable_id)
+        # Get the record
+        queryset = LocusGenotypeDisease.objects.filter(stable_id=g2p_stable_id, is_deleted=0)
+
+        if not queryset.exists():
+            self.handle_no_permission('Entry', stable_id)
+        else:
+            # Only 'undetermined' mechanisms can be updated
+            lgd_obj = queryset.first()
+            if lgd_obj.molecular_mechanism.mechanism.value != "undetermined":
```

Comment:
Suggesting that this is 
if lgd_obj.molecular_mechanism.mechanism.value != "undetermined" and lgd_obj.molecular_mechanism.mechanism_support.value != "inferred":

Reply:
I think to be clear it should be (Option A)
`if not (lgd_obj.molecular_mechanism.mechanism.value == "undetermined" and lgd_obj.molecular_mechanism.mechanism_support.value == "inferred"):`

The above expression is also equivalent to saying (Option B)
`if lgd_obj.molecular_mechanism.mechanism.value != "undetermined" or lgd_obj.molecular_mechanism.mechanism_support.value != "inferred":`

But Option A seems more clear that if it the mechanism and support combination is not ("undetermined" and "inferred") then we cannot update the mechanism

What are your thoughts?